File: surface/widgets/winui/components/widgets/domain/SegmentedProgressBar.cs
using Microsoft.UI.Xaml;
using Microsoft.UI.Xaml.Controls;
using Microsoft.UI.Xaml.Automation;

namespace Clef.Surface.Widgets.Concepts
{
    public sealed partial class SegmentedProgressBar : UserControl
    {
        private string _state = "idle";

        public SegmentedProgressBar()
        {
            this.InitializeComponent();
            AutomationProperties.SetName(this, "Horizontal progress bar divided into col");
        }

        public void Send(string eventType)
        {
            // State machine dispatch
        }
    }
}

File: surface/widgets/appkit/components/widgets/domain/SegmentedProgressBar.swift
import AppKit

class SegmentedProgressBarView: NSView {
    enum State: String { case idle; case animating; case segmentHovered }
    private var state: State = .idle

    override init(frame: NSRect) {
        super.init(frame: frame)
        setupSubviews()
        setAccessibilityRole(.group)
        setAccessibilityLabel("Horizontal progress bar divided into col")
    }

    required init?(coder: NSCoder) { super.init(coder: coder) }

    private func setupSubviews() {
        // Parts: root, bar, segment, segmentLabel, legend, totalLabel
    }

    func send(_ event: String) {
        // State machine dispatch
    }
}
